# This patch modifies CDE (https://github.com/lkujaw/cde), which is released
# under the GNU Lesser General Public License, version 2.0 or (at your option)
# any later version; see CDE's COPYING.  CDE's own terms require that
# modifications to it be made available under the GNU LGPL, so this patch is
# licensed under the same.
#
# SPDX-License-Identifier: LGPL-2.0-or-later
# Copyright (c) 2026 Justin Malcolm
# Derived from CDE, Copyright (c) 1993-2012 The Open Group.
# Modification: allow CDE_LOGFILES_TOP from the environment (dtspcd); 2026-10-07.
#
--- a/lib/DtSvc/DtEncap/nls.c
+++ b/lib/DtSvc/DtEncap/nls.c
@@ -36,17 +36,31 @@
 
 #include <bms/bms.h>
 #include <bms/MemoryMgr.h>
+#include <stdlib.h>	/* getenv, for the CDE_LOGFILES_TOP override below */
 
 #ifndef CDE_LOGFILES_TOP
 #define CDE_LOGFILES_TOP "/var/dt/tmp"
 #endif
 
+/* cde-wayland: the SPC daemon (dtspcd) redirects its stderr to
+ * CDE_LOGFILES_TOP/DTSPCD.log and refuses to start if that path is
+ * unwritable.  /var/dt/tmp is a system path an unprivileged Wayland session
+ * cannot create, so allow CDE_LOGFILES_TOP to name a writable directory. */
+static XeString
+_LogTop(void)
+{
+    XeString e = (XeString) getenv("CDE_LOGFILES_TOP");
+    if (e && *e)
+	return e;
+    return (XeString) CDE_LOGFILES_TOP;
+}
+
 /*------------------------------------------------------------------------+*/
 XeString
 XeSBTempPath(XeString suffix)
 /*------------------------------------------------------------------------+*/
 {
-    XeString dir   = (XeString) CDE_LOGFILES_TOP;
+    XeString dir   = _LogTop();
     XeString path;
 
     if (!suffix || !*suffix)
